packages/dev/core/src/node.ts

```diff
@@ -814,6 +814,38 @@ export class Node implements IBehaviorAware<Node> {
         return this._ranges[name] || null;
     }
 
+    /**
+     * Clone the current node
+     * @param name Name of the new clone
+     * @param newParent New parent for the clone
+     * @param doNotCloneChildren Do not clone children hierarchy
+     * @returns the new transform node
+     */
+    public clone(name: string, newParent: Nullable<Node>, doNotCloneChildren?: boolean): Nullable<Node> {
+        const result = SerializationHelper.Clone(() => new Node(name, this.getScene()), this);
+
+        result.name = name;
+        result.id = name;
+
+        if (newParent) {
+            result.parent = newParent;
+        }
+
+        if (!doNotCloneChildren) {
+            // Children
+            const directDescendants = this.getDescendants(true);
+            for (let index = 0; index < directDescendants.length; index++) {
+                const child = directDescendants[index];
+
+                if ((<any>child).clone) {

```

Comment:
Why do we need a `any` cast? `child` should be a `Node` type.

Reply:
Oh, this was copied from the TransformNode one so that's why it had this check 😅 thanks for catching!